manager/integration/tests/common.py

```diff
@@ -1684,6 +1684,33 @@ def wait_for_volume_migration_node(client, volume_name, node_id):
     return v
 
 
+def wait_for_volume_snapshot_purge_completion(client, volume_name):
+    completed = 0
+    last_purge_progress = {}
+    purge_status = {}
+    for i in range(RETRY_COUNTS):
+        completed = 0
+        v = client.by_id_volume(volume_name)
+        purge_status = v["purgeStatus"]
+        for status in purge_status:
+            assert status["error"] == ""
+
+            progress = status["progress"]
+            replica = status["replica"]
+            last = last_purge_progress.get(status["replica"])
```

Comment:
`last_purge_progress[status["replica"]]`?

Reply:
Returns a `KeyError` if it doesn't exist, which is what will happen the first time we poll the `Progress` since there's no `last_purge_progress` stored for any `Replicas`.

Reply:
Make sense.